### PR TITLE
Fix ACL Forms

### PR DESCRIPTION
**Pull request by turko**
FormBuilder expects a string and not an object

### CODE REVIEW

File: Util/AdminObjectAclManipulator.php [hunk 1]
```diff
@@ -272,7 +272,7 @@ protected function buildForm(AdminObjectAclData $data, FormBuilderInterface $for
                 );
             }
 
-            $formBuilder->add($key, new AclMatrixType(), array('permissions' => $permissions, 'acl_value' => $aclValue));
+            $formBuilder->add($key, AclMatrixType::class, array('permissions' => $permissions, 'acl_value' => $aclValue));
```
Comment (core23) [inline Util/AdminObjectAclManipulator.php]: This won't work with old PHP versions.

Comment (ju1ius) [inline Util/AdminObjectAclManipulator.php]: @core23 Isn't it the time to stop supporting 5.4? It's already long gone...

Comment (ju1ius) [inline Util/AdminObjectAclManipulator.php]: Bye bye array(array(array( :laughing: )))

Comment (core23) [inline Util/AdminObjectAclManipulator.php]: I am the last person who would support these old crap... But we use semver, so we must support it until the next major release.

**Comment by greg0ire** [comment]
@turko : would you like to make this in a BC way ? Or do I mark this as "major" and schedule it for 4.0 ?

File: Util/AdminObjectAclManipulator.php (hunk 1 above)
Comment (greg0ire) [inline Util/AdminObjectAclManipulator.php]: Don't use `::class` and this should be good to merge.

**Comment by turko** [comment]
Hi,
it require doctrine/common > 2.2 that is already requiring php> 5.5. To install this package with a lower would be not easy. Don't you think?
Here is a copy of the doctrine/commo composer
```
 "require": {
        "php": "~5.5|~7.0",
        "doctrine/inflector": "1.*",
        "doctrine/cache": "1.*",
        "doctrine/collections": "1.*",
        "doctrine/lexer": "1.*",
        "doctrine/annotations": "1.*"
    },
```

**Comment by turko** [comment]
Hi, thank you for the fast feedback. I change it to be compatible to php 5.3.

Greetings

File: Util/AdminObjectAclManipulator.php [hunk 2]
```diff
@@ -272,7 +271,7 @@ protected function buildForm(AdminObjectAclData $data, FormBuilderInterface $for
                 );
             }
 
-            $formBuilder->add($key, new AclMatrixType(), array('permissions' => $permissions, 'acl_value' => $aclValue));
+            $formBuilder->add($key, 'Sonata\\AdminBundle\\Form\\Type\\AclMatrixType', array('permissions' => $permissions, 'acl_value' => $aclValue));
```
Comment (greg0ire) [inline Util/AdminObjectAclManipulator.php]: Please linebreak this.

Comment (Soullivaneuh) [inline Util/AdminObjectAclManipulator.php]: Double `\` is not really necessary IMO and not used on the rest of the project.

**Comment by Soullivaneuh** [comment]
> Hi, thank you for the fast feedback. I change it to be compatible to php 5.3.

The goal is not only to make it compatible to php 5.3 but compatible with older version of Symfony.

**Comment by turko** [comment]
I add some line breaks and a check auf sf major_version. Greetings.

File: Util/AdminObjectAclManipulator.php [hunk 3]
```diff
@@ -271,8 +271,19 @@ protected function buildForm(AdminObjectAclData $data, FormBuilderInterface $for
                     'attr'     => $attr,
                 );
             }
-
-            $formBuilder->add($key, new AclMatrixType(), array('permissions' => $permissions, 'acl_value' => $aclValue));
+            if (3 < Kernel::MAJOR_VERSION) {
```
Comment (core23) [inline Util/AdminObjectAclManipulator.php]: Is the kernel version safe enough as a check?

Comment (Soullivaneuh) [inline Util/AdminObjectAclManipulator.php]: Absolutly not. This is from the Kernel component and we are working with the form one.

@turko Please use `method_exists`.

**Comment by Soullivaneuh** [comment]
Please also rebase to have StyleCI green.

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Comment by turko** [comment]
Hi Guys, sorry, but I don't see a way to do this check with out checking the kernel. I understand, that symfony forms are not depending on the kernel, but I didn't find a better way. We are talking about a breaking change in Symfony Forms that now expects a string where before it expected an Object (as a second param). I don't see how method_exists can help me on this one. I really believe that you should carefully evaluate if you want to support in version 3.x symfony 2.3.
I will do a rebase to get the styleCI.

**Comment by Soullivaneuh** [comment]
> I understand, that symfony forms are not depending on the kernel, but I didn't find a better way.

Here a sample of `method_exists` usage: https://github.com/sonata-project/SonataAdminBundle/blob/6555a6d0c6f5d97dcaa9b78f03f86f1cbdee7eaf/Form/Type/CollectionType.php#L30

> I really believe that you should carefully evaluate if you want to support in version 3.x symfony 2.3.

See discussion here: https://github.com/sonata-project/dev-kit/issues/51

> I will do a rebase to get the styleCI.

Don't forget to rebase and open another PR for `3.x` branch. :+1:

**Pull request closed without merging**